### ISSUE TITLE
Cannot drop partitions while inserting data using UDP sender

### ISSUE THREAD

**Issue by sherwinmascarenhas** [questdb/questdb#2460]
### Describe the bug

I have a service that is inserting data into quest using UDP ILP sender. As part of our data retention service we would like to drop partitions a few days old. However, when I run a simple drop partition command like the following using curl or through quest's UI: 

ALTER TABLE measurements DROP PARTITION LIST '2019-05-18';

I get a response saying:
```
JsonQueryProcessorState [179] waiting for update query
JsonQueryProcessor [fd=179] Resource busy, will retry
JsonQueryProcessorState [179] syntax-error [q=`ALTER TABLE '{table_name}' DROP PARTITION LIST '2022-08-15'`, at=0, message=`Query timeout. Please add HTTP header 'Statement-Timeout' with timeout in msQuery timeout. Please add HTTP header 'Statement-Timeout' with timeout in msTimeout expired on waiting for the async command execution result [instance=8]Timeout expired on waiting for the async command execution result [instance=9]Query timeout. Please add HTTP header 'Statement-Timeout' with timeout in msQuery timeout. Please add HTTP header 'Statement-Timeout' with timeout in msQuery timeout. Please add HTTP header 'Statement-Timeout' with timeout in msQuery timeout. Please add HTTP header 'Statement-Timeout' with timeout in ms`]
```

Increasing the timeout just makes the request run longer with the same 'Resource busy, will retry' error as above:
`cairo.writer.alter.busy.wait.timeout.micro=50000000` 

### To reproduce

1. Create quest instance that accepts udp messages on port 9009
2. Inserting metrics into the quest instance using the udp line protocol sender
3. Try to run drop partition commands while the insertion service runs in the background

### Expected Behavior

Drop partition command runs as excepted and does not constantly throw a resource busy error

### Environment

```markdown
- **QuestDB version**: 6.4.2
- **OS**: Debian GNU/Linux 11
- **Browser**:Chrome 104.0.5112.79 (Official Build) (arm64)
```

**Comment by sherwinmascarenhas** [comment]
Hi, 

Our table is partitioned by day. No the UDP metrics are not arriving out of order. 
The query we tried to use for dropping partitions is as follows: 
`ALTER TABLE 'table_name' DROP PARTITION WHERE ts < dateadd('d', -7, now())`
We also tried using an alternate query: 
ALTER TABLE 'table_name' DROP PARTITION LIST '2022-08-29';
But this did not work while we were persisting metrics using UDP. 

However both of these queries worked when we switched to peristing metrics to quest using TCP ILP.

**Comment by marregui** [comment]
Hi,

mmmm, it may be the version of QuestDB you are running. We recently worked on partition attach/detach and we revisited the drop functionality as well (version[ 6.5.1](https://github.com/questdb/questdb/releases/tag/6.5.1) released this). Don't do anything yet, I will test this scenario in both versions (yours and latest released 6.5.2) and will report back.

**Comment by sherwinmascarenhas** [comment]
Sure, that would be great!

**Comment by marregui** [comment]
Hi,

It is impossible in ILP/UDP as the table gets locked by the writer and is not released, because messages will arrive stochastically. 

You can do that with ILP/TCP

Would you be able to switch to ilp/tcp?

**Comment by sherwinmascarenhas** [comment]
Yes, we have switched back to ILP/TCP,  but we would prefer to do it in UDP, due to the higher rate of sending metrics and no waiting for ack. Would like to know if quest has plans to optimize the alter table process when using ILP/UDP or is it something that is not of priority.

**Comment by marregui** [comment]
It is not a priority at present, and I suppose it would require specking it out well, seems not so trivial.
Things as they are, QuestDB is running and listening for UDP packets. When one shows up, a table writer is created, the table becomes locked for write access and from that moment on the lock will exist and will prevent all write operations (which ALTER are).

What you can do is create a feature request, and then it will go through the standard flow of scheduling work on it as priorities shift.

Is TCP/UDP not giving you the performance you need?. We have recently released official clients, for instance https://github.com/questdb/py-questdb-client . This is highly optimised for ingress, it may be worth trying it. There are other clients for other languages too.

**Comment by sherwinmascarenhas** [comment]
Great! Will create a feature request for this as you've said. 
Regarding performance with TCP/ILP, we are using the latest clients and are seeing satisfiable performance. We are hoping the performance stays this way as we add more metrics to quest db. 

Thanks for all the help @marregui!
